### DISCUSSION TITLE
Breeze Cart page with Coupon code Apply and Cancel with Ajax

### DISCUSSION THREAD

**Discussion by niravGV** [breezefront/community#246] [2. Ideas]
Hello @vovayatsyuk 

We have already discussed this over another channel and you provide me below code for the functionality.

**Functionality: Need a Breeze cart page Coupon code apply and cancel with ajax and update the cart totals section like its happening with the based on shipping method selection** 

```js
$(document).on('submit', '#discount-coupon-form', async (e) => {
    var button = $('button.primary', e.target),
        input = $('#coupon_code', e.target),
        removeCoupon = $('#remove-coupon', e.target);

    e.preventDefault();

    if (button.hasClass('cancel')) {
        input.removeAttr('data-validate');
        removeCoupon.attr('value', '1');
    } else {
        input.attr('data-validate', '{required:true}');
        removeCoupon.attr('value', '0');
    }

    if (!$(e.target).valid()) {
        return;
    }

    button.spinner(true);

    await $.post({
        form: e.target,
        redirect: 'manual'
    });

    require([
        'Magento_Checkout/js/model/quote',
        'Swissup_Breeze/js/components/cart/estimation-services',
        'Magento_Customer/js/section-config'
    ], (quote, estimation, sections) => {
        sections.reload(['messages']);
        estimation.getTotals().then(result => {
            button.spinner(false);
            if (result.body.discount_amount) {
                button
                    .removeClass('apply')
                    .addClass('cancel')
                    .find('span')
                    .text($t('Cancel Coupon'))
            } else {
                input.val('').prop('disabled', false);
                button
                    .removeClass('cancel')
                    .addClass('apple')
                    .find('span')
                    .text($t('Apply Discount'))
            }
            quote.setTotals(result.body);
        });
    });
});
```

This code is not working with a breeze theme can you please help me to understand where the problem happening?

Thank you,

**Comment by vovayatsyuk** [comment]
The code works fine for me. How do you use it? Please provide the link to your site.

p.s. added a spinner to the code.

**Reply by niravGV** [reply]
Hello @vovayatsyuk 

This code works fine for me as well, and using into a custom.js file.

Thank you for the adding Spinner to your own given code..  :-)

I just want to ask that the bottom summery not visible discount which applied over the cart.

![image](https://github.com/breezefront/community/assets/30285024/0e53ba56-a244-446e-aa1c-707879733307)

![image](https://github.com/breezefront/community/assets/30285024/0d02182c-e6c3-4644-afd2-12d8cea08e4d)

@vovayatsyuk if i refresh the page still the coupon code which applied are not visible on a Cart summery section, and i am using default breeze theme and not having any additional module over that setup.

Thank you

**Reply by vovayatsyuk** [reply]
There is no discount returned in the response on your screenshot. I don't know why. Please check your discount rule in backend.

p.s. This doesn't happen to me. I do see the discount in the totals section.

**Reply by niravGV** [reply]
Hey @vovayatsyuk 

Thank you for the checking so quickly, you right some of the problem with a rule in backend.
Now, its working fine and looks good.

Thank you
